### DISCUSSION TITLE
blade_match

### DISCUSSION THREAD

**Discussion by rodgersjuan** [NAnand-TUD/parablade#11] [Q&A]
Hi,

I'm trying to run the blade_match in 2D

I've copied the config files from the examples to get a feel of how it works but I keep getting 
"    self.NDIM                       = int(self.IN["NDIM"][0])
TypeError: string indices must be integers"

my code is this:
my_blade = BladeMatch(**filepath of config file**)

Any help would be greatly appreciated!

**Comment by NAnand-TUD** [comment]
Hello !

So the argument of BladeMatch is not file path but the dictionary of options in the parablade configuration file.

Thanks,
Nitish

**Reply by rodgersjuan** [reply]
Hi Nitish,

Thanks for that.  I've amended my code like this now, similar to what was done in the examples. It's still not working, is tehre any chance you could upload an example .py file that shows how to set these correctly or just point me to how to change this? Thanks

It now throws this error "line 341, in get_DVs_functions
    nn = np.shape(P)[1]
IndexError: tuple index out of range"


my code:

section_variables = {}
section_variables['NDIM'] = [2]
section_variables['N_SECTIONS'] = [2]

section_variables['N_BLADES'] = [3]
section_variables['CASCADE_TYPE'] = 'LINEAR'
section_variables['PARAMETRIZATION_TYPE'] = "CAMBER_THICKNESS"
section_variables['OPERATION_TYPE'] = 'GEOMETRY'
section_variables['PLOT_FORMAT'] = 'MATPLOTLIB'
section_variables['x_leading'] = [0.]
section_variables['y_leading'] = [0.]
section_variables['z_leading'] = [1.00, 1.01]
section_variables['x_trailing'] = [1.]
section_variables['z_trailing'] =[1.00, 1.01]
section_variables['stagger'] = -45 * np.pi / 180
section_variables['theta_in'] = +30 * np.pi / 180
section_variables['theta_out'] = -70 * np.pi / 180
section_variables['radius_in'] = 0.05
section_variables['radius_out'] = 0.01
section_variables['dist_in'] = 0.60
section_variables['dist_out'] = 0.5
section_variables['thickness_upper_1'] = 0.15
section_variables['thickness_upper_2'] = 0.20
section_variables['thickness_upper_3'] = 0.13
section_variables['thickness_upper_4'] = 0.07
section_variables['thickness_upper_5'] = 0.03
section_variables['thickness_upper_6'] = 0.02
section_variables['thickness_lower_1'] = 0.15
section_variables['thickness_lower_2'] = 0.20
section_variables['thickness_lower_3'] = 0.13
section_variables['thickness_lower_4'] = 0.07
section_variables['thickness_lower_5'] = 0.03
section_variables['thickness_lower_6'] = 0.02
section_variables['x_hub'] = 0.0
section_variables['z_hub'] = 0.0
section_variables['x_shroud'] = 0.0
section_variables['z_shroud'] = 0.0
for i in section_variables:
    section_variables[i] = np.asarray(section_variables[i])

section_variables['PRESCRIBED_BLADE_FILENAME'] = 'C:/Users/Barnaby/Documents/Fortran/multall-open/MATLAB/para_blade.txt'

my_blade = BladeMatch(section_variables)

**Comment by JeoaFesketto** [comment]
Small clarification, because this isn't very obvious, there is a config.py file which contains a ReadUserInput(path_to_file) function to parse the config files given as example.
